### PR TITLE
4.13: OCPBUGS-11294: Add vSphere CSI migration to telemetry

### PR DESCRIPTION
**Pull request by jsafrane**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

Add `vsphere_csi_migration` to telemetry in 4.13.

WIP:
* ~Testing, Jira will follow soon.~ Tested OK with the PR below.
* Needs https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/146 merged first.

### CODE REVIEW

File: jsonnet/rules.libsonnet
Comment (jan--f) [inline jsonnet/rules.libsonnet:613]:
```suggestion
          expr: 'max by(status)vsphere_csi_migration{status=~"|LegacyDeprecatedInTreeDriver|CSIWithMigrationDriver"})',
```
Since we already know the possible label value set, lets be explicit here. I understand this is also constrained on the openshift/api level, but we'd like to establish this sort of aggressive cardinality specification for telemetry rules.

**Review by jan--f — CHANGES_REQUESTED** [review]
This looks good, just one request to make things easier for future readers.

File: jsonnet/rules.libsonnet
Comment (jsafrane) [reply jsonnet/rules.libsonnet:613]: Fixed

File: jsonnet/rules.libsonnet
Comment (jsafrane) [reply jsonnet/rules.libsonnet:614]: Yes, eventually. I created https://issues.redhat.com/browse/STOR-1289.

**Comment by jsafrane** [comment]
/jira refresh
/retest-required

I am not sure about ci/prow/versions - should I fix anything?

**Comment by jsafrane** [comment]
/hold
for https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/146 to merge

**Comment by jan--f** [comment]
> I am not sure about ci/prow/versions - should I fix anything?

No need, only the required checks need to succeed.

**Comment by jan--f** [comment]
/label backport-risk-assessed
/lgtm
Feel free to unhold once the blocking PR merged.

**Comment by simonpasquier** [comment]
/skip

the versions job isn't required.

**Comment by juzhao** [comment]
@jsafrane I am afraid you need to submit one PR to telemeter server, since "cluster:vsphere_csi_migration:max" is not in telemeter server allow list
https://github.com/rhobs/configuration/blob/main/configuration/telemeter/metrics.json
CMO PR is fine, add
/label cherry-pick-approved

**Pull request merged**
